modules/angular-meteor-collection.js

```diff
@@ -53,14 +54,61 @@ angularMeteorCollection.factory('AngularMeteorCollection', [
       return allPromise;
     };
 
-    AngularMeteorCollection.remove = function (keys) {
-      // remove collection
-      if (!keys) _.pluck(this, '_id');
-      // remove single doc
-      keys = [].concat(keys);
+    AngularMeteorCollection._upsertDoc = function(doc, useUnsetModifier) { 
```

Comment:
You're right, it should be `AngularMeteorCollection._upsertDoc.call(this, updates)`. I'm currently checking $meteorObject with a small sample app I built and I actually had some problems with that